monkestation/code/modules/replays/subsystem/replay.dm

```diff
@@ -23,125 +23,81 @@ SUBSYSTEM_DEF(demo)
 	var/last_written_time = null
 	var/last_chat_message = null
 
-	// stats stuff
-	var/last_queued = 0
-	var/last_completed = 0
-
-/datum/controller/subsystem/demo/proc/write_time()
-	var/new_time = world.time
-	if(last_written_time != new_time)
-		if(initialized)
-			WRITE_LOG_NO_FORMAT(GLOB.demo_log, "time [new_time]\n")
-		else
-			pre_init_lines += "time [new_time]"
-	last_written_time = new_time
-
-/datum/controller/subsystem/demo/proc/write_event_line(line)
-	write_time()
-	if(initialized)
-		WRITE_LOG_NO_FORMAT(GLOB.demo_log, "[line]\n")
-	else
-		pre_init_lines += line
+	var/disabled = FALSE
 
-/datum/controller/subsystem/demo/proc/write_chat(target, text)
-	var/target_text = ""
-	if(target == GLOB.clients)
-		target_text = "world"
-	else if(islist(target))
-		var/list/target_keys = list()
-		for(var/T in target)
-			var/client/C = CLIENT_FROM_VAR(T)
-			if(C)
-				target_keys += C.ckey
-		if(!target_keys.len)
-			return
-		target_text = jointext(target_keys, ",")
-	else
-		var/client/C = CLIENT_FROM_VAR(target)
-		if(C)
-			target_text = C.ckey
-		else
-			return
-	write_event_line("chat [target_text] [last_chat_message == text ? "=" : json_encode(text)]")
-	last_chat_message = text
+	// stats stuff
+	//var/last_queued = 0
+	//var/last_completed = 0
 
 /datum/controller/subsystem/demo/Initialize()
+#if defined(UNIT_TESTS) || defined(AUTOWIKI) // lazy way of doing this but idc
+	CONFIG_SET(flag/demos_enabled, FALSE)
+#endif
 	if(!CONFIG_GET(flag/demos_enabled))
-		flags |= SS_NO_FIRE
-		marked_dirty.Cut()
-		marked_new.Cut()
-		marked_turfs.Cut()
+		disable()
 		return SS_INIT_NO_NEED
 
-	var/rounder = file("[GLOB.demo_directory]/round_number.txt")
-	fdel(rounder)
-	WRITE_FILE(rounder, "[GLOB.round_id]")
-
 	WRITE_LOG_NO_FORMAT(GLOB.demo_log, "demo version 1\n") // increment this if you change the format
 	if(GLOB.revdata)
-		WRITE_LOG_NO_FORMAT(GLOB.demo_log, "commit [GLOB.revdata.originmastercommit || GLOB.revdata.commit]\n")
+		WRITE_LOG_NO_FORMAT(GLOB.demo_log, "commit [GLOB.revdata.commit || GLOB.revdata.originmastercommit]\n")
 
 	// write a "snapshot" of the world at this point.
 	// start with turfs
 	log_world("Writing turfs...")
 	WRITE_LOG_NO_FORMAT(GLOB.demo_log, "init [world.maxx] [world.maxy] [world.maxz]\n")
-	marked_turfs.Cut()
+	marked_turfs.len = 0
```

Comment:
`list.len = 0` is slightly faster than `list.Cut()` according to zewaka and kapu in coderbus